pkg/aggregator/time_sampler.go

```diff
@@ -124,7 +125,7 @@ func (s *TimeSampler) newSketchSeries(ck ckey.ContextKey, points []metrics.Sketc
 	return ss
 }
 
-func (s *TimeSampler) flushSeries(cutoffTime int64, series metrics.SerieSink) {
+func (s *TimeSampler) flushSeries(cutoffTime int64, series metrics.SerieSink, blocklist *utilstrings.Blocklist) {
```

Comment:
Rather than plumbing it through the entire flush call chain, would it make sense to store the blocklist directly in the time sampler, rather than the worker? That way we will need to refer to it only in two places.

Reply:
The `TimeSampler` isn't waiting on a select or anything, each one are actually managed by the TimeSamplerWorker. I'm not sure that it would decrease the size of the chain since it would be on the worker to pass the list to the sampler: AFAIU (only checked quickly) the worker would still have the plumbing to pass the list to the sampler, it would just be stored somewhere else. Please let me know if you think I should look more into this, I can address it in a separate PR.